src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -1015,6 +1015,14 @@ private SyntaxBase ObjectProperty()
                     return this.NewLine();
                 }
 
+                // Nested resource declarations may be allowed
+                if (allowResourceDeclarations && 
+                        (current.Type == TokenType.At || 
+                        (current.Type == TokenType.Identifier && current.Text == LanguageConstants.ResourceKeyword)))
+                {
+                    return Declaration();
+                }

```

Comment:
This might need a different approach. This doesn't use lookahead and so it's treating `resource` as a reserved name inside an object body at the top level.

Reply:
You may also have a property called `resource` in an object. Can probably workaround by enclosing the property name with quotes but lookahead would fix it as well. (Colon token following the identifier would be enough to distinguish.)

---
In reply to: [577216554](https://github.com/Azure/bicep/pull/1516#discussion_r577216554) [](ancestors = 577216554)